File: Sorting/Selection Sort/code.java
/*
 * 
 * Title: Selection Sort
 * 
 * Date: 1st June 2021
 * 
 */

import java.util.*;

public class code {

    public static int[] sSort(int[] arr, int n) {
        for (int i = 0; i < n; i++) {
            int min = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[min]) {
                    min = j;
                }
            }
            if (min != i) {
                int temp = arr[i];
                arr[i] = arr[min];
                arr[min] = temp;
            }
        }
        return arr;
    }

    public static void main(String[] args) throws java.lang.Exception {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];

        // Inputing the Array
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        sc.close();

        // Sorting
        arr = code.sSort(arr, n);

        // Printing the Array
        System.out.println("New Array: ");
        for (int i : arr) {
            System.out.print(i + " ");
        }
    }
}

// This code was implemented on day 26 of 100 days of code //

File: Sorting/Selection Sort/code.cpp
/*
 * 
 * Title: Selection Sort
 * 
 * Date: 30th August 2021
 * 
 */
#include<iostream>
using namespace std;

void sSort(int n , int arr[])
{
    for(int i=0 ; i<n-1 ; i++)
    {
        int sm = i+1;
        for(int j=i+1 ; j<n ; j++)
        {
            if(arr[j] < arr[sm])
            {
                sm = j;
            }
        }
        int temp = arr[i];
        arr[i] = arr[sm];
        arr[sm] = temp;
    }
    cout<<"Sorted Array: ";
    for(int i=0;i<n;i++)
    {
        cout<<arr[i]<<" ";
    }
}
int main()
{
    int n,k;
    cout<<"Enter the size of array: ";
    cin>>n;
    int arr[n];
    cout<<"Enter Array Elements: ";
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    sSort(n , arr);
    return 0;
}